docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+# Twoliter Design Document
+
+## Introduction
+
+Twoliter is a command line tool for building, testing and deploying custom variants of [Bottlerocket](https://github.com/bottlerocket-os/bottlerocket).
+
+### Background
+
+One of Bottlerocket's security features is its immutable root filesystem and lack of package manager.
+Because software cannot be installed on Bottlerocket after the image has been created, the build system supports creating different Bottlerocket *variants*.
+Variants allow for different software and settings to be defined in Bottlerocket at build time.
+
+Currently, in order to define a variant, it needs to be added to the `variants` directory of the Bottlerocket git repository.
+We call this *in-tree*, meaning that a variant can only be defined within Bottlerocket's git tree.
+Similarly, to add software to Bottlerocket, you need add it to the in-tree `packages` directory.
+
+A fundamental intention with variants is to allow for users to customize Bottlerocket for their unique use cases.
+To do so currently, however, requires forking and altering the Bottlerocket git repo.
+To stay up-to-date with core project would require frequent rebasing with conflicting changes in the core project.
+
+This document describes `twoliter`, a tool that will allow for the definition of a Bottlerocket variant externally from the main Bottlerocket project's git tree.
+Eventually `twoliter` will interface with another project underway to allow out-of-tree builds to define their own settings.
+Integration with the new settings system is out of scope for this document, and will be integrated with settings as both projects mature.
+
+### Glossary
+
+Here are a few of the terms we will use in this document:
+
+- **In-Tree**: Anything that exists in, or is defined by, the [main Bottlerocket git repository]
+- **Out-of-Tree Build (OOTB)**: User-customized builds of Bottlerocket defined by an external git repository owned by the user.
+- **Out-of-Tree Variants**: Synonymous with *Out of Tree Builds*.
+- **Builder**: A person who creates Bottlerocket variants using Twoliter.
```

Comment:
My find and replace was not comprehensive. Anywhere we see the word Builder it needs to be changed to Maintainer.